src/huffman_tree.rs

```diff
@@ -243,6 +244,13 @@ impl HuffmanTree {
                         }
                         index = -value as usize; // go to next node
 
+                        if index
+                            >= 2 * min(self.code_lengths_length as usize, Self::MAX_CODE_LENGTHS)
+                        {
```

Comment:
```suggestion
                        if index >= array.len() {
```

I think this is easier to read.